lib/src/main/kotlin/com/swmansion/starknet/data/types/StarknetByteArray.kt

```diff
@@ -4,16 +4,33 @@ import com.swmansion.starknet.data.types.conversions.ConvertibleToCalldata
 import com.swmansion.starknet.extensions.splitToShortStrings
 import com.swmansion.starknet.extensions.toFelt
 
+/**
+ * Represents a ByteArray struct from Cairo.
+ *
+ * The ByteArray struct is used to represent a string in Cairo.
+ *
+ * @param data The list of 31-byte chunks of the byte array
+ * @param pendingWord The last chunk of the byte array, which consists of at most 30 bytes
+ * @param pendingWordLen The number of bytes in [pendingWord]
```

Comment:
Should we add an `init` block verifying those restrictions in #436 or yolo?

Reply:
let's add them, better safe than sorry